.github/workflows/ci-build.yaml

```diff
@@ -176,6 +176,7 @@ jobs:
       E2E_ENV_FACTOR: 2
     strategy:
       fail-fast: false
+      max-parallel: 4
```

Comment:
each of the matrix tests runs on a different instance, so this won't help the problem. if you wanted to reduce the number of tests Go runs, you could modify the `-p` flag (it's currently set to `-p 20` in CI)